sdk/ai/Azure.AI.Projects/src/Custom/Datasets/Datasets.cs

```diff
@@ -17,76 +20,90 @@ public partial class Datasets
         /// <summary>
         /// Internal helper method to create a new dataset and return a BlobContainerClient to the dataset's blob storage.
         /// </summary>
-        private (BlobContainerClient ContainerClient, string OutputVersion) CreateDatasetAndGetContainerClient(string name, string inputVersion)
+        private (BlobContainerClient ContainerClient, string OutputVersion) CreateDatasetAndGetContainerClient(string name, string inputVersion, string? connectionName = null)
         {
+            var pendingUploadRequest = new PendingUploadRequest(
+                pendingUploadId: null,
+                connectionName: connectionName,
+                pendingUploadType: PendingUploadType.BlobReference,
+                serializedAdditionalRawData: null);
+
             var pendingUploadResponse = PendingUpload(
```

Comment:
nit: We can avoid unwrapping: `pendingUploadResponse.Value` if we will explicitly mark that we need to get `PendingUploadResponse`
```C#
PendingUploadResponse pendingUploadResponse = await PendingUploadAsync(
```